scripts/nvm_subscription/manager.py

```diff
@@ -321,3 +325,30 @@ def create_subscription(self, did: str, wallet_pvt_key: str, chain_id: int) -> D
         else:
             logger.error("Subscription purchase transaction failed")
             return {"status": "failed", "receipt": dict(receipt)}
+
+
+    def check_sufficient_sender_balance(self) -> Tuple[bool, str]:
+        w3 = self.web3
+        sender = self.sender
+        chain_id = w3.eth.chain_id
+
+        # For gnosis, native xdai is used for purchase 
+        if chain_id == 100:
```

Comment:
Can we make it a const, like a dictionary or something ? So we don't have magic numbers in the code ? Same for base.